lib/src/excel.dart

```diff
@@ -12,13 +12,36 @@ import 'util.dart';
 /// Service class that provides dart functions to interact with Excel.
 @Injectable()
 class ExcelDart {
-  static const _startAddress = 'A1';
-  static final _startAddressInt = 'A'.codeUnitAt(0);
+  static const _tableColumnStartAddress = 'A';
+  static const _tableRowStartAddress = 4;
+  static const _thresholdAddress = '\$B2';
+
   static const _fontName = 'Roboto';
   static const _fontSize = 12;
-  static const _horizontalAlignment = 'Center';
+
+  static const _centerAlignment = 'Center';
+  static const _rightAlignment = 'Right';
+  static const _leftAlignment = 'Left';
+
   static const _borderStyle = 'Continuous';
+
   static const _currencyFormat = '\$#,##0.00';
+  static const _underpacingFormatFont = '#9C0006';
+  static const _underpacingFormatFill = '#FFC7CE';
+
+  /// Column addresses calculated based on [_tableHeader].
+  static final _spentColumn =
+      '\$${_getExcelColumnReference(_tableHeader.indexOf('Spent'))}'
+      '${_tableRowStartAddress + 1}';
+  static final _budgetColumn =
+      '\$${_getExcelColumnReference(_tableHeader.indexOf('Budget'))}'
+      '${_tableRowStartAddress + 1}';
+  static final _startDateColumn =
```

Comment:
is this ok if the table doesnt exist yet (i.e. is it evaluated lazily?)

Reply:
Seems like so? it didn't complain or throw an error